### PR TITLE
Aman3003 patch 3

### PR DESCRIPTION
**Pull request by AMAN3003**
PR FOR THE @mhvk 

FIX: #4550 Explicitly set BeautifulSoup parser to html.parser to avoid warning
and  fix for the Ipython Notebook and EarthLocation representation warning error #4542

### CODE REVIEW

**Comment by MSeifert04** [comment]
@AMAN3003 - Could you include a test for the Ipython Notebook failure? Just to see that things are working? I just tried your solution and it raises another error:

```
from astropy.coordinates import EarthLocation
from astropy.time import Time
a = EarthLocation.of_site('Paranal Observatory')
a

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
C:\Programming\Anaconda\lib\site-packages\IPython\core\formatters.py in __call__(self, obj)
    339             method = _safe_get_formatter_method(obj, self.print_method)
    340             if method is not None:
--> 341                 return method()
    342             return None
    343         else:

C:\Programming\Anaconda\lib\site-packages\astropy\units\quantity.py in _repr_latex_(self)
   1030                                               style=Latex.format_exponential_notation,
   1031                                               max_line_width=np.inf,
-> 1032                                               separator=',~')
   1033                 latex_value = latex_value.replace('...', r'\dots')
   1034             finally:

C:\Programming\Anaconda\lib\site-packages\numpy\core\arrayprint.py in array2string(a, max_line_width, precision, suppress_small, separator, prefix, style, formatter)
    439         if isinstance(x, tuple):
    440             x = _convert_arrays(x)
--> 441         lst = style(x)
    442     elif reduce(product, a.shape) == 0:
    443         # treat as a null array if any of shape elements == 0

C:\Programming\Anaconda\lib\site-packages\astropy\units\format\latex.py in format_exponential_notation(cls, val)
    104         """
    105         if np.any(np.isfinite(val)):
--> 106             m, ex = utils.split_mantissa_exponent(val)
    107 
    108             parts = []

C:\Programming\Anaconda\lib\site-packages\astropy\units\format\utils.py in split_mantissa_exponent(v)
     63     mantissa, exponent : tuple of strings
     64     """
---> 65     x = "{0:.8g}".format(v).split('e')
     66     if x[0] != '1.' + '0' * (len(x[0]) - 2):
     67         m = x[0]

TypeError: non-empty format string passed to object.__format__

<EarthLocation (1946618.261025452, -5467645.090548764, -2642488.447714588) m>
```

Also I'm not sure if you should alter ``astropy.units.format.latex.py`` in this case because this is working correct. I think the root of the problem is in ``astropy.coordinates.EarthLocation`` or ``astropy.units.Quantity``.

**Comment by AMAN3003** [comment]
@MSeifert04 how should i add test for ipython ? could you guide 
@MSeifert04  My commit is also not showing in the community page could you help me out why it's so

**Comment by MSeifert04** [comment]
@AMAN3003 - Maybe Actually a test for ipython notebooks is not so easy but I think you could just directly call ``astropy.coordinates.EarthLocation._repr_latex_`` since that's where the error starts. Or is your question more general about how to write a test at all?

What do you mean by community page?

**Comment by AMAN3003** [comment]
@MSeifert04  yes you are write how write test at all?
and by community page I means this https://github.com/astropy/astropy/graphs/contributors
even though i have made  commits

**Comment by AMAN3003** [comment]
in my case no warning is raised in  ipython after doing the fix

**Comment by MSeifert04** [comment]
@AMAN3003 : You are Using Ipython from the command line and not the Ipython Notebook. These are quite different things! Please try to call ``_repr_latex_`` and see if it raises an error.

About the tests, since this would make a unit test astropy uses the [pytest](http://pytest.org/latest/) and good examples for tests can be found in the astropy source code in the ``tests`` folder of each subpackage.

Only if your PR is *merged* you will be on that page, I guess. Never seen it before but the first line on the page suggests it.

**Comment by AMAN3003** [comment]
screenshot from 2016-02-05 00 00 06
@MSeifert04  after the above fix I am not getting any warning.

**Comment by MSeifert04** [comment]
Sorry, you want help and guidance but it seems you don't even read my comments.

From Ipython and with pytest you need to call ``a._repr_latex_()`` directly. Because ``Ipython NOTEBOOK`` calls this method when trying to print the representation. See also https://github.com/astropy/astropy/pull/4566#issuecomment-179971477 and https://github.com/astropy/astropy/pull/4566#issuecomment-179993666 where I already mentioned this.

**Comment by MSeifert04** [comment]
@AMAN3003 Also please read through https://github.com/astropy/astropy/issues/4542#issuecomment-179178326 and https://github.com/astropy/astropy/issues/4542#issuecomment-179339467. Because I think you are trying to fix something that's not broken while the underlying issue really comes from somewhere else.

**Comment by eteq** [comment]
@MSeifert04 is right about why it's not on the community page - you have to get a PR merged first.

Also, @AMAN3003, in general it's a bad idea to address two issues on the same PR unless they're actually realted.  That way if it turns out one has a problem or something, it won't delay the other.

**Comment by AMAN3003** [comment]
@eteq  Sorry sir i will not repeat it in future.
I am not able to merge PR beacuse of this travis ci error could you tell how to fix this travis ci error?
https://travis-ci.org/astropy/astropy/builds/106905685

**Comment by bsipocz** [comment]
> I am not able to merge PR beacuse of this travis ci error could you tell how to fix this travis ci error?

@AMAN3003  - Could you please read the development guideline? It was linked many times before, https://astropy.readthedocs.org/en/stable/development/workflow/development_workflow.html. 

1) You cannot merge a PR as only a few core maintainers has merge rights. And even then, all PRs have to be reviewed by at least one different person than the author before it can be merged.

2) Travis is failing for this PR, and actually gives rather good feedback about the problems:
```
astropy/io/ascii/html.py:22:3: E111 indentation is not a multiple of four

astropy/io/ascii/html.py:24:3: E111 indentation is not a multiple of four

astropy/io/ascii/html.py:26:3: E111 indentation is not a multiple of four

astropy/io/ascii/html.py:94:15: E111 indentation is not a multiple of four

astropy/io/ascii/html.py:96:15: E111 indentation is not a multiple of four

5
```

**Comment by AMAN3003** [comment]
@bsipocz  thank you mam it really worked. could  you review my work .

**Comment by AMAN3003** [comment]
@MSeifert04  Sorry Sir for ignoring your previous comment i am re working on this issue and would fix it as soon as possible

**Comment by AMAN3003** [comment]
@MSeifert04  I install the ipython notebook .and run the changed script this was the output.I think so this error is only occuring in dependencies of anaconda package 
please could you tell why i am not seeeing the warning
screenshot from 2016-02-06 11 29 24
screenshot from 2016-02-06 11 28 54

**Comment by MSeifert04** [comment]
@AMAN3003 - Actually I don't know why your Notebook isn't displaying LaTeX, it's been a long time since I've setup my notebook and it might not be the default. Maybe someone else knows this. But you can always verify the error by explicitly calling ``a._repr_latex_()``.

File: astropy/io/ascii/html.py [hunk 1]
```diff
@@ -83,7 +90,10 @@ def process_lines(self, lines):
                                         'installed to read HTML tables')
 
         if 'parser' not in self.html:
-            soup = BeautifulSoup('\n'.join(lines))
+            if lxml_installed:
+                soup = BeautifulSoup('\n'.join(lines),"lxml")
+            else:
+                soup = BeautifulSoup('\n'.join(lines),"html.parser")
```
Comment (embray) [inline astropy/io/ascii/html.py]: Incidentally this patch is *almost* fine modulo some stylistic bits:
1) For consistency's sake, rename `lxml_installed` to `HAVE_LXML`
2) Add spaces after commas between function arguments.

And then rebase or whatever to remove the unrelated change in `astropy.units`.  Also re-title the PR to what it's actually addressing.

**Comment by taldcroft** [comment]
Closed by Astropy Project Coordinators.
